src/transport/RendezvousSession.cpp

```diff
@@ -179,7 +179,9 @@ void RendezvousSession::OnSessionEstablished()
     {
         ChipLogError(Ble, "Missing node id in rendezvous parameters. Node ID is required until opcerts are implemented");
     }
-    mPairingSession.PeerConnection().SetPeerNodeId(mParams.GetRemoteNodeId().ValueOr(kUndefinedNodeId));
+
+    const auto defaultPeerNodeId = mParams.IsController() ? kTestDeviceNodeId : kTestControllerNodeId;
```

Comment:
This seems to hardcode node ids.
Is there some other workaround here - this is very odd.

Reply:
I believe I had to preserve the peer node id during message sending on rendezous so I would expect the remote node id to be available during communication.

Could we use that instead of having this IsController?

in any case this requires a large 'TODO: this needs replacement' if it actually goes in.

Reply:
@andy31415 Did you intentionally remove "source node id" from messages sent by the commissioner? If so, there's currently no other way to initialize it properly other than hard coding. Once we have CASE sessions implemented, we may no longer need to know for which node ID a given encryption key was added, but now we do need this information.

Reply:
Actually, I can see that you moved setting source Node ID to `RendezvousSession`, so in theory `PASESession` could set `mConnectionState.SetPeerNodeId` on receipt of the session establishment message...

Reply:
I thought the whole point of removing the source IDs from the PASE code paths is that in PASE world, node IDs are not authenticated, and so much less meaningful.  Wouldn't the 'correct' thing to do here be to store the session encryption keys in the session object itself and pass this around?

It seems wrong to index the keys on Node ID when anybody can claim any Node ID during PASE.  The only thing that is authenticated is that a peer, whomever they are, at the end of some known channel (BLE, peer/peer IP tuple or whatever), are in possession of a shared secret.  Using the session object would seem to key to this more directly and could avoid some odd DOS situations (or worse) where another party also claims during PASE to have the same node ID.

Reply:
You may be right, however if you look into the current implementation of `SecureSessionMgr::OnMessageReceived` you will see that the key used for decryption of the message is selected based on the source node ID from that message. So until we don't align this part with the spec we must ensure that the `nodeID -> connection state/encryption key` mapping is correctly initialized and that is not the case. I'm OK either with merging my quick fix or reverting the changes which break the previous behavior and working out a better solution, but the point is that the issue is blocking some devs and test autiomation engineers, so we should somehow address it quickly :)

Reply:
Reverting the other PR seems like a move in the wrong direction.

Thanks for the explanation. Looks good to me.